package/base-files/files/lib/upgrade/common.sh

```diff
@@ -91,7 +91,7 @@ get_image() { # <source> [ <command> ]
 		esac
 	fi
 
-	cat "$from" 2>/dev/null | $cmd
+	$cmd "$from"
```

Comment:
Hi!
This change breaks default_do_upgrade functionality:
```bash
# Flash firmware to MTD partition
#
# $(1): path to image
# $(2): (optional) pipe command to extract firmware, e.g. dd bs=n skip=m
default_do_upgrade() {
	sync
	if [ -n "$UPGRADE_BACKUP" ]; then
		get_image "$1" "$2" | mtd $MTD_ARGS $MTD_CONFIG_ARGS -j "$UPGRADE_BACKUP" write - "${PART_NAME:-image}"
	else
		get_image "$1" "$2" | mtd $MTD_ARGS write - "${PART_NAME:-image}"
	fi
	[ $? -ne 0 ] && exit 1
}
```
$cmd is supposed to read image file from stdin, process it, and output processed firmware through stdout. It's a pipe command and doesn't take source image as an argument at all.
This cat isn't redundant. If you have other reason to change this behavior please write them down in commit message.

Reply:
I see, this is not the behavior on x86, where depending on the image, the $cmd is determined. I did not check other platforms. I'll drop it.